test/end-to-end/test_studio_with_ssl_cert_file_envvar_set.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Test that SSL_CERT_FILE is persisted into the studio and 
+# set to the correct internal path. 
+
+set -euo pipefail
+
+cleanup_cached_certificate() {
+  local hab_ssl_cache
+  hab_ssl_cache="/hab/cache/ssl"
+
+  rm -f "$hab_ssl_cache/*"
+}
+
+studio_run() {
+  cleanup_cached_certificate "$e2e_certname"
+  hab studio run "$@"
+}
+
+echo "--- Generating a signing key"
+hab origin key generate $HAB_ORIGIN
+
+echo "--- Generating self-signed ssl certificate"
+tempdir="$(mktemp --tmpdir --directory e2e-ssl-XXXXXX)"
+e2e_certname="e2e-ssl.pem"
+openssl req -newkey rsa:2048 -batch -nodes -keyout key.pem -x509 -days 365 -out "${tempdir}/${e2e_certname}"
+
+export SSL_CERT_FILE="${tempdir}/${e2e_certname}"
+
+echo "--- SSL_CERT_FILE is correctly set inside the studio"
```

Comment:
@scotthain  I'm not certain that multiple assertions in the same e2e test is the right pattern, splitting these out would make it easier to see what is being tested, but you'd end up with file and pipeline step sprawl.  In addition, I _think_ we'll want to run these same sets of tests across the different methods of entering a Linux studio, which would necessitate splitting up setup from the test (for Windows w/Linux containers)

Do you have any advice on we should be structuring these sorts of tests?